docs/specifications/tests/Connectivity-TP/connectivity02.md

```diff
@@ -1,48 +1,196 @@
-the domain are accessible over TCP on port 53.
+This test case will verify if the name servers in the delegation of and zone of
+*Child Zone* are reachable over TCP.
+
+This test case will mimic the tests done by [Connectivity01], but over TCP
+instead:
+
+* Name Server responding to a query.
+* Name Server including SOA record of *Child Zone* in the answer section in the
+  response on a SOA query for *Child Zone*.
+* Name Server including NS record of *Child Zone* in the answer section in the
+  response on an NS query for *Child Zone*.
+* Name Server setting the AA flag in a response with SOA or NS in answer section.
+* Name Server responding with expected [RCODE Name] ("NoError") on query for SOA
+  or NS for *Child Zone*.
 
-## Inputs
 
-The domain to be tested.
+## Scope
 
-## Ordered description of steps to be taken to execute the test case
+The only TCP port defined for DNS is port 53 ([RFC 1035][RFC 1035#4.2.1], section
+4.2.1), and that is the only port used by this and other test cases for DNS
+queries to the name servers.
+
+UPD connectivity is tested by test case [Connectivity01].
+
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+
+## Summary
+
+Message Tag                         |Level  |Arguments  | Message ID for message tag
+:-----------------------------------|:------|:----------|:---------------------------------------------------------------------------------------
+CN02_MISSING_NS_RECORD_TCP          |WARNING| ns        | Nameserver {ns} reponds to a NS query with no NS records in the answer section over TCP.
+CN02_MISSING_SOA_RECORD_TCP         |WARNING| ns        | Nameserver {ns} reponds to a SOA query with no SOA records in the answer section over TCP.
+CN02_NO_RESPONSE_NS_QUERY_TCP       |WARNING| ns        | Nameserver {ns} does not respond to NS queries over TCP.
+CN02_NO_RESPONSE_SOA_QUERY_TCP      |WARNING| ns        | Nameserver {ns} does not respond to SOA queries over TCP.
+CN02_NO_RESPONSE_TCP                |WARNING| ns        | Nameserver {ns} does not respond to any queries over TCP.
+CN02_NS_RECORD_NOT_AA_TCP           |WARNING| ns        | Nameserver {ns} does not give an authoritative response on an NS query over TCP.
+CN02_SOA_RECORD_NOT_AA_TCP          |WARNING| ns        | Nameserver {ns} does not give an authoritative response on an SOA query over TCP.
+CN02_UNEXPECTED_RCODE_NS_QUERY_TCP  |WARNING| ns, rcode | Nameserver {ns} responds with an unexpected RCODE ({rcode}) on an NS query over TCP.
+CN02_UNEXPECTED_RCODE_SOA_QUERY_TCP |WARNING| ns, rcode | Nameserver {ns} responds with an unexpected RCODE ({rcode}) on an SOA query over TCP.
+CN02_WRONG_NS_RECORD_TCP            |WARNING| ns        | Nameserver {ns} responds with a wrong owner name ({owner} instead of {name}) on NS queries over TCP.
+CN02_WRONG_SOA_RECORD_TCP           |WARNING| ns        | Nameserver {ns} responds with a wrong owner name ({owner} instead of {name}) on SOA queries over TCP.
```

Comment:
Can be removed. Those are not defined argument names (yet?).
Although we could add them if we wanted, if so then also specify them as output in steps 5.3.1.4 and 5.3.2.4.

Reply:
Corrected argument name and updated in steps 3.3.1.4 and 3.3.2.4.